.github/workflows/tests.yml

```diff
@@ -0,0 +1,61 @@
+name: Tests
+on:
+  push:
+  pull_request:
+  pull_request_target:
+
+jobs:
+
+  tests:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ] # , macos-latest, windows-latest ]
+        python-version: [ "3.6", "3.7", "3.8", "3.9" ]
+        django-version: [ "2.2", "3.1" ]
+        cryptography-version: [ "3.1.0", "3.2.0", "3.3.0", "3.4.0" ]
+
+    env:
+      DJANGO_CA_SECRET_KEY: dummy
+
+    name: Python ${{ matrix.python-version }}, Django ${{ matrix.django-version }}, cryptography ${{ matrix.cryptography-version }}, OS ${{ matrix.os }}
+    steps:
+
+      - name: Acquire sources
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: x64
+
+      - name: Apply caching of dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: os=${{ matrix.os }}-python=${{ matrix.python-version }}-pip-${{ hashFiles('**/requirements-*.txt') }}
+
+      - name: Install designated Django and cryptography versions
+        run: |
+          pip install Django~=${{ matrix.django-version }}
+          pip install cryptography~=${{ matrix.cryptography-version }}
+
+      - name: Install dependencies
+        run: |
+          pip install -r requirements/requirements-core.txt
+          pip install -r requirements/requirements-libs.txt
+          pip install -r requirements/requirements-test.txt
+          pip install -r requirements/requirements-docs.txt
```

Comment:
Hmm, one second. ACTUALLY I just recently noticed the same on travis. You *first* install Django/cryptography, and then the requirements. This means that dependencies  will install different versions again. So we're not testing what we think we're testing. If you look closely, that is already happening. `requirements-core.txt` will install a different version of cryptography!

So you need to invert the two stages (install reqs, install django/cryptograph).

Reply:
Hi again.

I believe it will do no harm because the constraint `cryptography>=2.8` in `requirements-core.txt` will already be completely satisfied.

Proof:
```sh
# Install test matrix requirements.
pip uninstall cryptography
pip install cryptography~=3.1.0
Collecting cryptography~=3.1.0
  Using cached cryptography-3.1.1-cp35-abi3-macosx_10_10_x86_64.whl (1.8 MB)
Successfully installed cryptography-3.1.1

# Install package requirements.
pip install -r requirements/requirements-core.txt
[...]
Requirement already satisfied: cryptography>=2.8 in ./.venv/lib/python3.6/site-packages (from -r requirements/requirements-core.txt (line 3)) (3.1.1)
[...]
python -c 'import cryptography; print(cryptography.__version__)'
3.1.1
```

With kind regards,
Andreas.

Reply:
> I believe it will do no harm because the constraint cryptography>=2.8 in requirements-core.txt will already be completely satisfied.

I disprove:

```
Run pip install -r requirements/requirements-core.txt
...
Successfully installed ... cryptography-3.4.6
```

See e.g. [this build](https://github.com/mathiasertl/django-ca/runs/1943697881?check_suite_focus=true)

Reply:
Hi Mathias,

I see. Bummer! Thanks for fixing it with fb0fdf0. Does it work now?

With kind regards,
Andreas.